ramalama/ollama.py

```diff
@@ -30,15 +30,18 @@ def pull_blob(repos, layer_digest, accept, registry_head, models, model_name, mo
     layer_blob_path = os.path.join(repos, "blobs", layer_digest)
     url = f"{registry_head}/blobs/{layer_digest}"
     headers = {"Accept": accept}
-    download_file(url, layer_blob_path, headers=headers, show_progress=True)
-
-    # Verify checksum after downloading the blob
-    if not verify_checksum(layer_blob_path):
-        print(f"Checksum mismatch for blob {layer_blob_path}, retrying download...")
-        os.remove(layer_blob_path)
+    local_blob = in_existing_cache(model_name, model_tag)
+    if local_blob is not None:
```

Comment:
In the:

`if local_blob is not None:`

case, we seem to call `ln -sf` twice, should we call it once?

In the rm case in an Ollama-stored object, I propose we just remove the symlink part:

local_blob

Reply:
I agree, I think we should only remove the ramalama content on `ramalama rm`.  But on `ollama rm` ramalama can end up in a bad state as well, which I think we should take care of also.  `ramalama ls` should either ignore hanging symlinks or remove them. 

BTW I have seen the database get futz up by breaking removing of images, and we don't handle it cracefully.  I had to find the symlink and rm -f it .

Reply:
Actually after thinking through this, I understand the double symlink now, once for the sha256 and a second time for the named version, I think the rm should handle this without touching the Ollama store, but we should test to be sure.